### PR TITLE
working for latex files with % comments or \begin{comment}...\end{com…

### PR DESCRIPTION
**Pull request by alexhagen**
Should work now (I tested on a simple .tex file) for .tex files, using either comments like this:

```latex

% TODO: a comment here

\begin{comment}
  TODO: or a comment like this that can go across multiple lines
  and still be together
\end{comment}
```

### CODE REVIEW

**Comment by pgilad** [comment]
Thanks! Can you fix the broken build (I think it's lint) and un-delete `docs/REPORTERS.md`?

**Comment by alexhagen** [comment]
Sure, I'll get to it this weekend, if not sooner.

**Comment by alexhagen** [comment]
Hello,

I'm having a little trouble getting the checks to pass on Travis.  When I run them on my system (OSX, npm v6.3.0, eslint v5.4.0, prettier v1.14.2, mocha v5.2.0), I get

```bash
hage581$ npm run test

> leasot@6.3.0 test /Users/hage581/code/leasot
> mocha -R spec ./tests/*.js

...

  reporting
    vscode
      ✓ typescript
      ✓ reference-leading
      ✓ edge-cases


  84 passing (3s)
```

and
```bash
hage581$ npm run prettier-check

> leasot@6.3.0 prettier-check /Users/hage581/code/leasot
> prettier --list-different '{bin,lib}/**/*.js' 'tests/*.js'
```
(no output), and

```bash
hage581$ npm run lint

> leasot@6.3.0 lint /Users/hage581/code/leasot
> eslint .
```

I'm not very familiar with Travis, so I'm not sure why the tests are failing.  The error from Travis is

```bash
npm ERR! node v6.14.3
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! leasot@6.3.0 prettier-check: `prettier --list-different '{bin,lib}/**/*.js' 'tests/*.js'`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the leasot@6.3.0 prettier-check script 'prettier --list-different '{bin,lib}/**/*.js' 'tests/*.js''.
npm ERR! Make sure you have the latest version of node.js and npm installed.
```

The only thing I can think of is the big discrepancy between node/npm versions on my machine and the travis config (10.5.0/6.3.0 versus 6.14.3/3.10.10).  Any ideas?

File: .travis.yml [hunk 1]
```diff
@@ -11,6 +11,7 @@ matrix:
   fast_finish: true
 script:
   - npm run lint
+  - npm run prettier
```
Comment (pgilad) [inline .travis.yml]: Doing this will obviously let the prettier check pass 😄 . Without committing the changes it really defeats the purpose. Run `npm run prettier` on your PR branch and commit the changes (if any)

**Comment by pgilad** [comment]
Hey this looks awesome, can you run `npm run prettier` on your code and see if there's any diff? If so commit that and see if travis still fails

**Comment by alexhagen** [comment]
```bash
hage581$ npm run prettier

> leasot@6.3.0 prettier /Users/hage581/code/leasot
> prettier --write '{bin,lib}/**/*.js' 'tests/*.js'

bin/leasot-reporter.js 37ms
bin/leasot.js 28ms
lib/cli-reporter.js 29ms
lib/cli.js 35ms
lib/parsers.js 47ms
lib/parsers/coffeeParser.js 12ms
lib/parsers/defaultParser.js 17ms
lib/parsers/ejsParser.js 11ms
lib/parsers/erlangParser.js 8ms
lib/parsers/hamlParser.js 15ms
lib/parsers/haskellParser.js 6ms
lib/parsers/hbsParser.js 9ms
lib/parsers/jadeParser.js 7ms
lib/parsers/latexParser.js 15ms
lib/parsers/pascalParser.js 14ms
lib/parsers/pythonParser.js 7ms
lib/parsers/ssParser.js 7ms
lib/parsers/twigParser.js 8ms
lib/reporters.js 8ms
lib/reporters/custom.js 17ms
lib/reporters/json.js 3ms
lib/reporters/markdown.js 9ms
lib/reporters/raw.js 2ms
lib/reporters/table.js 16ms
lib/reporters/vscode.js 9ms
lib/reporters/xml.js 3ms
lib/utils/comments.js 10ms
lib/utils/index.js 2ms
tests/cli-spec.js 39ms
tests/logic-spec.js 9ms
tests/parser-spec.js 117ms
tests/reporter-spec.js 9ms

hage581$ git status
On branch master
Your branch is up to date with 'origin/master'.

nothing to commit, working tree clean
hage581$ git diff
hage581$
```

Looks like no diff - so nothing to commit.

I don't think the `npm run prettier` change you described above is obvious. Is that test local?  All tests worked locally, then not on Travis. They work if you run `prettier` before `prettier-check`, so I wrote that in to travis, which you seem to be opposed to.

I'm closing this pull request and will just use my own fork.  Feel free to use any of the code in my commits and integrate them into `leasot` on your fork.

**Comment by pgilad** [comment]
Let me have a look at your branch

**Comment by pgilad** [comment]
Sorry for the time it took to review @alexhagen .. I'm on vacation 🙈 

Will be handled in #102

**Comment by pgilad** [comment]
@alexhagen Also, added auto-prettier/eslint on git commits so hopefully people won't run into this problem again ;)

https://github.com/pgilad/leasot/pull/103
